instrumentation/net/http/otelhttp/internal/semconv/env.go

```diff
@@ -110,10 +117,13 @@ func (s HTTPServer) RecordMetrics(ctx context.Context, md ServerMetricData) {
 
 	attributes := oldHTTPServer{}.MetricAttributes(md.ServerName, md.Req, md.StatusCode, md.AdditionalAttributes)
 	o := metric.WithAttributeSet(attribute.NewSet(attributes...))
-	addOpts := []metric.AddOption{o}
-	s.requestBytesCounter.Add(ctx, md.RequestSize, addOpts...)
-	s.responseBytesCounter.Add(ctx, md.ResponseSize, addOpts...)
+	addOpts := metricAddOptionPool.Get().(*[]metric.AddOption)
```

Comment:
Shouldn't we empty the slice here?

Reply:
I empty the slice before I put it back in the pool. I didn't want to be redundant, but if it is better to empty it again after getting it from the pool, that sounds good to me.

Reply:
I updated the code in case it was better practice, but I can revert if needed. Let me know if it looks good.

Reply:
Ah right, sorry, I missed it. Before or after doesn't change much IMHO (with your latest commit, you're doing it both before and after though).

Reply:
Ok, I will do it after since I am guessing it is less ideal to have objects with different states in the pool.